sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java

```diff
@@ -105,49 +95,54 @@
      * Current schema.
      */
     private static final ContentValues SCHEMA =
-            getContentValues("", "", new Document<>(), "", 0, 0, 0, "");
+            getContentValues("", "", "", "", 0, 0, 0, "");
 
     private final DatabaseManager mDatabaseManager;
 
-    private LocalDocumentStorage(DatabaseManager databaseManager) {
-        mDatabaseManager = databaseManager;
+    LocalDocumentStorage(Context context, String userTable) {
+        mDatabaseManager = new DatabaseManager(context, DATABASE, READONLY_TABLE, VERSION, SCHEMA, new DatabaseManager.DefaultListener());
+        if (userTable != null) {
+            createTableIfDoesNotExist(userTable);
+        }
     }
 
-    LocalDocumentStorage(Context context) {
-        this(new DatabaseManager(context, DATABASE, TABLE, VERSION, SCHEMA, new DatabaseManager.DefaultListener(), new String[]{PARTITION_COLUMN_NAME, DOCUMENT_ID_COLUMN_NAME}));
+    /**
+     * Creates a table for storing user partition documents.
+     */
+    void createTableIfDoesNotExist(String userTable) {
```

Comment:
Is it based on another PR? I see a lot of changes from #1056

Reply:
Need to change branch target?

Reply:
Re-targeted with #1056 branch.